source/d3d8to9_device.cpp

```diff
@@ -1060,6 +978,7 @@ HRESULT STDMETHODCALLTYPE Direct3DDevice8::GetPaletteEntries(UINT PaletteNumber,
 }
 HRESULT STDMETHODCALLTYPE Direct3DDevice8::SetCurrentTexturePalette(UINT PaletteNumber)
 {
+	PaletteNumber = (PaletteNumber & 0xFFF) | (PC_NOCOLLAPSE << 12);
```

Comment:
I am not quite sure.  I did some research and the `PALETTEENTRY` structure is documented [here](https://msdn.microsoft.com/en-us/library/windows/desktop/bb147253(v=vs.85).aspx).  It seems to crash anytime `SetCurrentTexturePalette` is called with any flag set other than `PC_NOCOLLAPSE`.  However it is not clear why it crashes.  I did find the comment below in the documentation, but that is all.  These flags do not seem well documented:

> The alpha intensity value for the palette entry. Note that as of DirectX 8, this member is treated differently than documented for Windows.